### ISSUE TITLE
A lot of "UNKNOWN" and missing civilization grouping

### ISSUE THREAD

**Issue by claudiusmueller** [robertjanetzko/LegendsBrowser2#12]
Fantastic job on the rewrite. Loading files is MUCH faster than before. I'm just listing this issue here to help find the weak points.

Using LegendsBrowser 2.0.5 and DF steam 50.09.

Using an xml file exported from the steam version I find many "UNKNOWN POSITION", "UNKNOWN OCCASION", "UNKNOWN JOB" listings (for example when selecting "The Trade of Warmth" dwarven civilization) in this file (https://www.dropbox.com/scl/fi/z4sh7dc3tsaujvsfk9v1r/region2-00250-01-01-legends.xml?rlkey=ir785g7u7aroh9y4t3e0z4x22&dl=0).

In addition, civilizations are not sorted/filtered correctly. Only "kobold" and "necromancer" civilizations are grouped, the rest are listed as "unknown".

**Comment by nicomitchell** [comment]
This is tested on Steam version on Windows.
Example of an unknown historical event:
```xml
	<historical_event>
		<id>73797</id>
		<type>change_hf_state</type>
		<hfid>14561</hfid>
		<state>settler</state>
		<reason>none</reason>
		<site>1357</site>
	</historical_event>
```

Renders as 
`In late winter of 102, UNKNWON HistoricalEventChangeHfState`


Shows in regular legends mode as `In the <season> of <year>, the <race> <name> settled in <site name>`

**Comment by Galliad93** [comment]
The error log refers to some kind of indexoutofbounds error. Also it seams not to be able to find any files, when I load the exported xml file. 

Loading: C:\Program Files (x86)\Steam\steamapps\common\Dwarf Fortress\region1-00161-01-01-legends.xml
157.09 MiB / 157.09 MiB [-----------------------------------------------------------------] 100.00% 140.75 MiB p/s 1.3s

no legends_plus.xml found

no worldgen params found

no world map found

no world history found

processing...
world ready

this is what the console tells me.

**Comment by robertjanetzko** [comment]
Not all event data is in the export of the base version. If you want to see everything you need to export the legends data with dfhack. An error that occurred when unknown events are loaded is fixed in version 2.0.6
